### PR TITLE
Detail a list of available pytest plugins to perform parallel exploration

### PR DESCRIPTION
**Pull request by andfoy**
See https://github.com/Quansight-Labs/free-threaded-compatibility/issues/86#issuecomment-2453485039

### CODE REVIEW

File: docs/debugging.md [hunk 1]
```diff
@@ -127,6 +127,23 @@ PYTHON_GIL=0 pytest -x -v --count=100 test_concurrent.py
 We advise to set `count` in the order of hundreds and even larger, in order to
 ensure at least one concurrent clash event.
 
+## PyTest plugins to discover concurrency issues
```
Comment (lysnikolaou) [inline docs/debugging.md:130]:
```suggestion
## pytest plugins to discover concurrency issues
```

File: docs/debugging.md [hunk 2]
```diff
@@ -127,6 +127,23 @@ PYTHON_GIL=0 pytest -x -v --count=100 test_concurrent.py
...
+
+Given that the aforementioned testing scenarios are repetitive and highly
+verbose, several community-lead pytest libraries have been implemented
```
Comment (lysnikolaou) [inline docs/debugging.md:133]:
```suggestion
verbose, several community-led pytest plugins have been implemented
```

File: docs/debugging.md [hunk 3]
```diff
@@ -127,6 +127,23 @@ PYTHON_GIL=0 pytest -x -v --count=100 test_concurrent.py
...
+including,
+
+- [pytest-run-parallel](https://github.com/Quansight-Labs/pytest-run-parallel)
+- [pytest-freethreaded](https://github.com/tonybaloney/pytest-freethreaded).
+
+The advantage of using a pytest plugin as opposed to manually using the testing
+scenarios mainly reside in their ability to integrate with the ecosystem
```
Comment (lysnikolaou) [inline docs/debugging.md:140]:
```suggestion
scenarios mainly resides in their ability to integrate with the ecosystem
```

**Review by lysnikolaou — COMMENTED** [review]
Thanks @andfoy! Generally looks good to me, with just a couple of things to fix.

**Review by rgommers — COMMENTED** [review]
I think the section on pytest plugins should come first, since that's the better way to get started. As written now, doing the same thing manually is probably only useful to:

1. test only a small subset of functionality with concurrent execution,
2. being able to run some concurrent tests without having to add a new test dependency

Anything else? It would be good to explain why that large and complex code block is shown (i.e., the reasons above), and if the rationale for it is quite thin then consider hiding it in a foldout or in a note/info block where it gets rendered in smaller font.

File: docs/debugging.md (hunk 1 above)
Comment (ngoldbaum) [inline docs/debugging.md:130]: I'd also group the discussion about `pytest-repeat` into this section.

**Review by rgommers — APPROVED** [review]
Pushed a few copy-edits, LGTM now. Thanks @andfoy & reviewers!

**Pull request merged**